### DISCUSSION TITLE
Attempting to run express with lighthouse-ci ssl cert auth with no joy

### DISCUSSION THREAD

**Discussion by bradyma1** [GoogleChrome/lighthouse-ci#656] [Q&A]
attempting to run express-wrapper.js
with the following parameters - 
const express = require('express')
const lhci -= require('@lhci/server')
const fs = require ('fs')
const https = require('https')
const settings = requires('lighthouserc.js')

(async  () => {
      
       const app = express();
      console.log(settings.ci.server)
      const (app: lhciApp) = await lhci.createApp(settings.ci.server);

      const cert = fs.readFileSync('./server.cert.pem')
      const key = fs.readFileSync('./server,key.pem');
      const ca = fs/readFileSync('./ca-bundle.pem');

     const httpsServer = https.createServer({cert, key, ca}, app);

    app.usr(lhciApp);
    app.listen(8443);
debugger}090;

_______________________________________________________________________________

symptom - This site cannot provide a secure connection  ERR_SSL_PROTOCOL_ERROR
netstat -a LISTEN |grep 8443 shows app is listening properly
If I try with http.....I immediately see the debug log moving and I can log in successfully.
certs are in docker file and I can see them in the docker instance
Lighthouserc.js ( as before)  is working and connecting properly to the database
I am using lhci/server and cli 6.8.1 and lhci/utils 0.3.8

Thanks in advance for any suggestions

**Comment by patrickhulce** [comment]
In general, I'd recommend using a reverse proxy like nginx or haproxy with https in front of node rather than handling TLS in the LHCI server directly if it's going to be handling any external traffic.

It sounds like this is a docker container already, perhaps a slightly modified lhci-server with this setup could work nicely?

**Comment by bradyma1** [comment]
You are correct.  It is a docker container...
I already have a lighthouse.js the has an ssl client connection to postgres as follows - 
output of lighthouserc.js
_______________________
const fs = require('fs');

module.exports = {
     ci: {
       server: {
          storage: {
             storageMethod: ' sql',
             sqlDialect: ' postgres',
             sqlConnection Ssl: true
             sqlConnectionUrl: 'postgres://cert@x.xxx.xx.xxx/lhci1'
             sqlDialecrtOptions: {
                ssl: {
                   ca: fs.readFileSync('.ca.pem', 'utf8'),
                   key: fs.readFileSync('.client.key', 'utf8'),
                   cert: fs.readFileSync('.client.pem', 'utf8'),
                  },
            },
         },
       },
    },
   }
   

______________________________________________________

Are you proposing that I modify this to create the https server dialogue I need for ssl authentication to the lhci server?

**Reply by patrickhulce** [reply]
> Are you proposing that I modify this to create the https server dialogue I need for ssl authentication to the lhci server?

No, I was not. And now I'm not sure what you're trying to accomplish. 

Do you just want an SSL connection from LHCI to the database? 
Or do you want an SSL connection from client to LHCI server?

If it's the former, you don't need to be touching the express app at all.
If it's the latter, see my previous suggestion about nginx. The solution there is well beyond the scope of LHCI-specific help.

**Comment by bradyma1** [comment]
Sorry, I misunderstood. 

I am looking for SSL connection from client to LHCI server.  I will check nginx.  For some reason, we thought you were recommending express previously.  My bad I will research this.,
